bindings/cs/rl.net/AsyncSender.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Rl.Net.Native;
+
+namespace Rl.Net
+{
+    public abstract class AsyncSender : ISender
+    {
+        private ErrorCallback errorCallback;
+
+        public AsyncSender(ErrorCallback callback)
+        {
+            this.errorCallback = callback;
+        }
+
+        public virtual void Init(ApiStatus status)
+        {
+        }
+
+        public void Send(SharedBuffer buffer, ApiStatus status)
+        {
+            // Create a cloned handle so that we have an our own copy of the shared pointer to the buffer 
+            SharedBuffer ownedHandle = new SharedBuffer(buffer);
+            GC.KeepAlive(buffer);
+
+            Task backgroundTask = SendAsyncAndUnwrapExceptions(ownedHandle);
```

Comment:
I tried here to capture the same semantics as in [eventhub_client.cc](https://github.com/VowpalWabbit/reinforcement_learning/blob/master/rlclientlib/logger/eventhub_client.cc#L126): It creates a task (though it manually manages the max number of tasks, rather than relying on the async/await system to manage this) and fire-and-forgets it.

The idea would be that it is possible to implement a sync sender, as well as a FNF one like the native EventHub one.